app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    result: dict = {}
+
+    @wraps(func)
+    def inner(*args, **kwargs) -> int:
+        if args not in result.keys():
+            result[args] = func(*args, **kwargs)
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+        return result[args]
```

Comment:
as a key in result you should be using not only args, because in this case - we are ignoring kwargs, so your cache will not work properly. Please, fix it. Try to use args and kwargs as key. Consider, that kwargs is a dict, so it can't be directly used as key.

Reply:
Can you check my changes?